src/react/connect.ts

```diff
@@ -0,0 +1,29 @@
+import { useStream, NOT_YET_EMITTED } from './useStream';
+import { Observable } from 'rxjs';
+import React, { ComponentType } from 'react';
+
+/**
+ * Higher order component for connecting a component to a stream
+ *
+ * The stream does not need to provide all the properties the component expects,
+ * any missing properties will be required by the component returned from this
+ * component.
+ *
+ * @param WrappedComponent Component that will recieve props from the stream
+ * @param stream$ Stream that will provide props to the component
+ * @see useStream
+ */
+export const connect = <Props, Observed extends Partial<Props>>(
+  Component: ComponentType<Props>,
+  stream$: Observable<Observed>
+) => (props: Omit<Props, keyof Observed>) => {
+  const value = useStream(stream$);
```

Comment:
I think I remember you mentioned that this would solve `connectWithProps` as well, do you have an example of that?

Reply:
I'm purely guessing here, but I think Tobias' intention might have been to replace `connectWithProps` with the HoC from this PR. That would explain the order in which props vs streamedProps are spread. However this is (potentially) not backwards-compatible with `connect`.

Reply:
`connectWithProps` use properties as a part of the stream (usually filtering). The `useStream` hook can be used in it's place.
This HOC cannot replace `connectWithProps` because the properties are not available when the stream is initialized.

I'll create a PR in ardoq-front where I replace `connectWithProps` with `useStream` today.